torch/nn/parallel/data_parallel.py

```diff
@@ -1,10 +1,67 @@
+import weakref
 import torch
+import torch.cuda.comm as comm
+from collections import OrderedDict
 from ..modules import Module
 from .scatter_gather import scatter_kwargs, gather
 from .replicate import replicate
 from .parallel_apply import parallel_apply
 
 
+# NOTE: these callbacks are one-off
+# NOTE: they are dedupliacted based on identity
+class _CallbackOrderedDict(OrderedDict):
```

Comment:
This class needs an explanation of what it does.

Reply:
If python-core says that this is a good way to implement this, we should post an update to https://stackoverflow.com/questions/26189090/python-detect-if-any-element-in-a-dictionary-changes :)

Reply:
Yeah I saw multiple threads regarding that and almost none of them was particularly helpful. It's not sufficient to override `__setitem__` to get full coverage. Not sure who do you mean as python-core.